### PR TITLE
Properly set Vboxed field during union codegen

### LINKED ISSUE #44921: Illegal instruction crash on 1.7
I experienced a crash due to `signal (4): Illegal instruction`:
```
Unreachable reached at 0x7f91eb7569e5

signal (4): Illegal instruction
in expression starting at /home/salmon/Documents/code/YetAnotherTestFramework/test/v2/order_trait_test.jl:37
OrderTrait at /home/salmon/Documents/code/YetAnotherTestFramework/src/v2/executor/parse_time_trait.jl:68
unknown function (ip: 0x7f91e7f235cf)
Allocations: 8889567 (Pool: 8885928; Big: 3639); GC: 11
```

I have uploaded an rr trace of this crash [here](https://s3.amazonaws.com/julialang-dumps/reports/2022-04-09T18-29-17-Octogonapus.tar.zst).

versioninfo:
```
Julia Version 1.7.2
Commit bf53498635 (2022-02-06 15:21 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, skylake)
Environment:
  JULIA_PKG_SERVER = 
  JULIA_EDITOR = code --wait
```

This also happens on the latest commit
```
Julia Version 1.9.0-DEV.353
Commit 73c9863087 (2022-04-09 15:06 UTC)
```

### PR DESCRIPTION
**Pull request by ianatol**
Fixes #44921 

All credit to @vtjnash and @maleadt for the [solution](https://github.com/JuliaLang/julia/issues/44921#issuecomment-1095303549) and [test case](https://github.com/JuliaLang/julia/issues/44921#issuecomment-1094651878) respectively, all I did was package into this PR.

### CODE REVIEW

File: test/compiler/codegen.jl [hunk 1]
```diff
@@ -702,6 +702,23 @@ function f42645()
...
+end
+function f44921(a)
+    if a == :x
+        A44921(_f) # _f purposefully undefined
+    elseif a == :p
+        g44921(a)
+    end
+end
+function g44921(a)
+    if !any([])
```
Comment (vtjnash) [inline test/compiler/codegen.jl:716]: this can be any unknown condition, so might be simplest to make it `_f` also?

Comment (ianatol) [reply test/compiler/codegen.jl:716]: Since this code path is actually run, not sure how we can use `_f` here without getting an `UndefVarError`?

Comment (vtjnash) [reply test/compiler/codegen.jl:716]: ah, true, would need to be something like `!@isdefined _f`

**Review by vtjnash — APPROVED** [review]
SGTM

**Comment by DilumAluthge** [comment]
The `Random` failure on the `x86_64-linux-gnu` tester is https://github.com/JuliaLang/julia/issues/42752

```
The global RNG seed was 0x79104eec59565e99cd1b518b8970892.
 
Error in testset Test:
Test Failed at /cache/build/default-amdci5-7/julialang/julia-master/julia-72691483d9/share/julia/stdlib/v1.9/Test/test/runtests.jl:795
  Expression: orig == Random.default_rng()
   Evaluated: Xoshiro(0xb085b7d5b1c6a426, 0x7128c07ac362f37f, 0x86fa3d51e90d640c, 0x65b8e7a89ffd023f) == TaskLocalRNG()
Error in testset Test:
Test Failed at /cache/build/default-amdci5-7/julialang/julia-master/julia-72691483d9/share/julia/stdlib/v1.9/Test/test/runtests.jl:796
  Expression: rand(orig) == rand()
   Evaluated: 0.436257938130729 == 0.8544673546079519
ERROR: LoadError: Test run finished with errors
```

**Pull request merged**